quinn/src/runtime.rs

```diff
@@ -266,7 +266,7 @@ mod tokio;
 #[cfg(feature = "runtime-tokio")]
 pub use self::tokio::TokioRuntime;
 
-#[cfg(feature = "async-io")]
```

Comment:
Might be better to cfg out `async_io::UdpSocket` specifically? The `AsyncTimer` impl could, theoretically, be useful on its own, and so long as we have a freestanding async-io feature we might as well use it for that.

Might also be best to place this fix in a separate PR so we can merge it independently.

Reply:
In addition to that, even though the `async_io` module exists with only the `async-io` feature, it won't be exported, because the `pub use self::async_io::*` line below is feature-gated to the smol and async-std feature.
I don't think this makes sense, these two cfgs should just agree.

So. I can't *not* do changes here, as this breaks CI. But any other change doesn't seem sensical to me (and would be much bigger diffs). I'd argue this is the best fix?

Reply:
> I don't really understand why it's needed for this PR, but not for main

That does seem strange. Maybe the feature testing is randomized? Maybe something in the toolchain updated since the last run on main?

> Given it causes a CI failure, I'm not 100% sure we can merge this independently.

I figured we'd merge it *first*, since with the proposed tweak it'd be non-breaking, and then this PR can be rebased and merged whenever we're ready to break.

> the pub use self::async_io::* line below is feature-gated to the smol and async-std feature.

We don't actually need to re-export anything for the trait impl on `Timer` to be available downstream. While it's a pretty niche hypothetical case where that's useful, masking it out is technically avoidable breakage.

Reply:
I have now rebased onto main after #2264